### PR TITLE
fix: builtins.html does replace [hash] and [name] in builtins.html.filename

### PR DESCRIPTION
**Pull request by suxin2017**
## Related issue (if exists)

[builtins.html does replace [hash] and [name] in builtins.html.filename](https://github.com/web-infra-dev/rspack/issues/2638)

## Types of changes

- [ ] Docs change / Dependency upgrade
- [x] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run changeset`.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by ahabhgk — APPROVED** [review]
Thanks!

**Comment by ahabhgk** [comment]
`pnpm run changeset` is needed for fix

**Comment by suxin2017** [comment]
> `pnpm run changeset` is needed for fix

I'm not sure I did the right thing. Please check

**Review by ahabhgk — DISMISSED** [review]

File: crates/rspack_plugin_html/src/plugin.rs [hunk 1]
```diff
@@ -172,8 +179,38 @@ impl Plugin for HtmlPlugin {
...
+        .file_name()
+        .and_then(OsStr::to_str)
+        .map(|str| {
+          str
+            .split('.')
+            .collect::<Vec<_>>()
+            .first()
+            .expect("TODO:")
+            .to_string()
+        }),
```
Comment (ahabhgk) [inline crates/rspack_plugin_html/src/plugin.rs:195]:
```suggestion
      name: Path::new(&url).file_stem().map(|s| s.to_string_lossy().to_string()),
```

**Review by ahabhgk — CHANGES_REQUESTED** [review]

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**